packages/venia-concept/src/components/Pagination/pagination.js

```diff
@@ -167,10 +167,12 @@ class Pagination extends Component {
 
         const queryPage = Math.max(
             1,
-            ~~getQueryParameterValue({
-                location,
-                queryParameter: 'page'
-            })
+            Math.floor(
```

Comment:
This change is completely unrelated to the rest of this PR but [since I had to spend time looking up what the ~~ operator does](https://stackoverflow.com/questions/5971645/what-is-the-double-tilde-operator-in-javascript), I thought I'd save all future readers the effort.

Reply:
It *is* faster. Maybe a comment instead :)

Reply:
You looked it up and now you know what it does. No need to change it. 👍

Reply:
So I dug a little deeper and [wrote up a little something](https://codepen.io/supernova_at/post/javascript-s-double-bitwise-not-operator).

Turns out that `~~` is not a straight replacement for `Math.floor`, specifically for the case of negative floating point numbers.

So on one hand since we're pulling our input from the query string, we can't be certain that it will be a positive number. But on the other hand, negative page numbers don't make sense (let alone negative floating point page numbers) so I don't think we care about them.

So I reverted this change and added a comment, as @sirugh suggested.

I do want to voice my opinion though, that we should strongly consider readability when we write our code - especially in Venia, which is supposed to be a reference implementation.

Is `~~` faster? Sure. Is it more concise? Yeah. Do either of those things matter here? I'd argue that they do not.

If `~~` was a widely understood and commonly used paradigm in JavaScript I'd have less of a problem, but my gut tells me that the majority of the people who run across and try to understand this code are going to have to look up double bitwise NOT, and we can make this much more developer friendly.